engine/values/values.go

```diff
@@ -98,6 +116,14 @@ func (v *Values) SetOutput(stepName string, value interface{}) {
 	v.setStepData(stepName, OutputKey, value)
 }
 
+// SetPreHook stores a step's prehook in Values
+func (v *Values) SetPreHook(output, metadata interface{}) {
+	v.m[PreHookKey] = map[string]interface{}{
```

Comment:
I don't think so. We initialise it because in the other function we directly access the map without checking and instantiating the map if not exist. This function SetPreHook instantiate every time the map and we have no access to the keys (except in text/template that manage it very well by itself)